client/state/comments/reducer.js

```diff
@@ -21,172 +21,146 @@ import {
 	COMMENTS_REQUEST_SUCCESS,
 	COMMENTS_REQUEST_FAILURE,
 	DESERIALIZE,
-	SERIALIZE,
+	SERIALIZE
 } from '../action-types';
 import {
-	getCommentParentKey,
-	updateExistingIn
-} from './utils';
-import {
 	PLACEHOLDER_STATE
 } from './constants';
 
-/***
- * Creates a function that updates like count
- * @param {Boolean} like true to like, false to unlike
- * @returns {Function} function that updates like count for comment
- */
-function getLikeSetter( { like = true, likeCount } ) {
-	return ( comment ) => comment.set( 'i_like', like )
-								.update( 'like_count', ( count ) => {
-									if ( typeof likeCount !== 'undefined' ) {
-										return likeCount;
-									}
-
-									return like ? count + 1 : count - 1;
-								} );
-}
-
-/***
- * Updates a specific state in the general state
- * @param {Immutable.Map} state general state for all commentTargetId
- * @param {Object} action action object
- * @param {Number} action.siteId site identifier
- * @param {Number} action.postId post identifier
- * @param {Function|Object} updaterOrValue an updater function or a value
- * @returns {Immutable.Map} new state
- */
-function updateSpecificState( state, action, updaterOrValue ) {
-	const id = getCommentParentKey( action.siteId, action.postId );
-
-	if ( typeof updaterOrValue === 'function' ) {
-		return state.update( id, ( value ) => updaterOrValue( value ) );
+const updateComment = ( commentId, newProperties ) => comment => {
+	if ( comment.ID !== commentId ) {
+		return comment;
 	}
+	const updateLikeCount = has( newProperties, 'i_like' ) && ! has( newProperties, 'like_count' );
```

Comment:
I didn't wanted to change the original logic TOO much. There should be a `like_count` (I think), but the original function did: https://github.com/Automattic/wp-calypso/pull/12382/files#diff-512900ef67147a63a42ecc06c3abcdbbL46